### ISSUE TITLE
Usage with SSR returns "Missing translationId: *ID* for language: ${ languageCode }"

### ISSUE THREAD

**Issue by ramyma** [ryandrewjohnson/react-localize-redux#125]
**Do you want to request a _feature_ or report a _bug_?**
Report a bug

**What is the current behavior?**
When rendering with `ReactDOMServer.renderToString`, the translations are not ready at the time of page render and therefore yields the uninitialized message of the translation. 

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code. Paste the link to your JSFiddle (https://jsfiddle.net/Luktwrdm/) or CodeSandbox (https://codesandbox.io/s/new) example below:**
Here's an example:
https://codesandbox.io/s/2z9vqwwxlp

**What is the expected behavior?**
The localization should be initialized before render time to get the translated message rendered correctly by the server.

**Which versions of `react` and `react-localize-redux` are you using?**
v3.3.2

**Comment by ryandrewjohnson** [comment]
This is probably related to issue #103

**Comment by ramyma** [comment]
I'm not using redux and I thought it deserves a new thread as it's not very much related..

I suppose we need to have a way to initialize the provider with some props to avoid having the context uninitialized at the time of SSR.

@ryandrewjohnson  What's your thoughts ?

**Comment by ryandrewjohnson** [comment]
Regardless of redux you need to use LocalizeProvider. The related issue I believe is that LocalizeProvider is dependent on componentDidMount lifecycle method that doesn’t get called for SSR. I need to investigate whether the logic in componentDidMount can be moved to the constructor instead, as that would get called both in SSR and browser rendering.

**Comment by ramyma** [comment]
Correct me if I'm wrong, the code you're referring to in cdm is 
```
componentDidMount() {
    if (this.props.store) {
      this.unsubscribeFromStore = storeDidChange(
        this.props.store,
        this.onStateDidChange.bind(this)
      );
    }
  }
```
Which only gets called if a store is passed to LocalizeProvider.
So I suppose it actually differs if redux is not used, no ?

**Comment by ryandrewjohnson** [comment]
Ah you’re right. If you have some time to investigate that would be great. I still don’t have an example mocked for SSR so will be some time before I can get to this.

**Comment by ramyma** [comment]
I have a theory, if we can pass the initialize object to the LocalizeProvider and then use the passed values as the defaults when setting up the context, we'd be able to get the context initialize right from the start. 
I've been doing some digging in the code but I don't know where would that fit exactly.. 
I'd like to help, can you give me any pointer ?

BTW I've created a minimalistic library example [here](https://codesandbox.io/s/2z9vqwwxlp) to test SSR with the

**Comment by ramyma** [comment]
@ryandrewjohnson what do you think ?

**Comment by ryandrewjohnson** [comment]
Fixed in #117 and included in version `v3.4.0`.
